docs/connector-development/config-based/connector-builder-compatibility.md

```diff
@@ -0,0 +1,210 @@
+```
+curl -X POST  -H "Content-Type: application/json"  -d '{"username": "person@metabase.com", "password": "fakepassword"}'    http://localhost:3000/api/session
+```
+
+Use the session token to fetch the data
+
+```
+curl -X GET \
+-H "Content-Type: application/json" \
+-H "X-Metabase-Session: 38f4939c-ad7f-4cbe-ae54-30946daf8593" \
+http://localhost:3000/api/user/current
+```
+
+
+Example: [Metabase](https://www.metabase.com/learn/administration/metabase-api#authenticate-your-requests-with-a-session-token)
+
+### Other
+AWS endpoints are examples of APIs requiring a non-standard authentication mechanism. You can tell from [the documentation](https://docs.aws.amazon.com/pdfs/awscloudtrail/latest/APIReference/awscloudtrail-api.pdf#Welcome) that requests need to be signed with a hash.
+
+Example: [AWS Cloudtrail](https://docs.aws.amazon.com/pdfs/awscloudtrail/latest/APIReference/awscloudtrail-api.pdf#Welcome)
+
+If the integration requires a non-standard authentication mechanism, use Python CDK or low-code with custom components. If the integration uses OAuth2.0, verify if it uses single-use refresh tokens.
+
+## Is the data returned as JSON format?
+Is the data returned by the API formatted as JSON, or is it formatted as another format such as XML, CSV, gRPC, or PDF?
+
+Examples:
+- Yes: [Congress API](https://api.congress.gov/)
+- No: [Federal Railroad Administration (FRA) Safety Data APIs](https://safetydata.fra.dot.gov/MasterWebService/FRASafetyDataAPIs.aspx)
+
+If the data is not formatted as JSON, use the Python CDK.
+
+## How are records paginated?
+Look up the pagination mechanism in the API documentation, and identify which type it is.
+
+Here are the standard pagination mechanisms:
+
+### Page count
+Endpoints using page count pagination accept two pagination parameters
+1. The number of records to be returned (typically called “page_size”)
+2. The page to request (typically called “page”)
+
+Example: [newsapi.ai](https://newsapi.ai/documentation)
+
+![Page-count-example](./assets/connector_builder_compatibility/page_count_example.png)
+
+### Limit-Offset (OffsetIncrement)
+Endpoints using limit-offset pagination accept two pagination parameters
+1. The number of records to be returned (typically called “limit”)
+2. The ID of the first record to return (typically called “offset”)
+
+![Offset-limit-example](./assets/connector_builder_compatibility/offset_limit_example.png)
+
+Example: [Congress API](https://api.congress.gov/)
+
+### Link to next page (CursorPagination)
+Endpoints paginated with a link to the next page of records typically include either a “Link” field in the response header, or in the response body.
+
+You can search the documentation and the sample response for the “next” keyword.
+
+Example: [Greenhouse](https://developers.greenhouse.io/harvest.html#pagination)
+
+### Does the integration require query params that are not key-value pairs?
+Do requests require complex query parameters? Does it support GraphQL querying?
```

Comment:
Not really sure what to do about these questions - if I answer yes, what does it mean? If I answer no, what does it mean? Should be clearly answered in the very next sentence. Especially for graphql it's not clear if you don't know what graphql is / how it works.

Reply:
rephrased. let me know if this is clearer